Add an option to use the mouse's location as an anchor when zooming

samuellapointe opened the pull request:
By default, zooming in or out zooms using the mouse's location as an anchor, which makes it possible to navigate the map using scrolling only, and makes zooming in more intuitive. 

This patch adds a new `useAnchor` option to the MouseWheelZoom interaction. It is enabled by default, which changes nothing to the current behavior of the interaction. When disabled, the interaction stops registering its lastAnchor_ value. This turns the zoom into a very basic one, that zooms to the center of the screen. 

The reason I am creating this patch is for a @mapgears project involving tracking points as they move on the map. We want to follow a point as it moves on the map, and stop the user from moving the map around as the point is followed. Leaving the zoom enabled would let the user move around using the mouse scroll, which could be confusing if they get lost and can't come back using panning.

A good example is [this demo](http://openlayers.org/en/master/examples/geolocation-orientation.html) on the openlayers website, where the same issue was solved by relocating to the point after zooming in or out.

adube commented on externs/olx.js:
```diff
@@ -2647,6 +2647,14 @@ olx.interaction.MouseWheelZoomOptions.prototype.duration;
+ * @type {boolean|undefined}
+ * @api
+ */
+olx.interaction.MouseWheelZoomOptions.prototype.useAnchor;
```
Add this new property in the options definition, above.

elemoine replied:
It would be good to document the behavior when this option is set to `false`.

elemoine commented on src/ol/interaction/mousewheelzoominteraction.js:
```diff
@@ -119,3 +129,20 @@ ol.interaction.MouseWheelZoom.prototype.doZoom_ = function(map) {
+ */
+ol.interaction.MouseWheelZoom.prototype.enableMouseAnchor = function() {
+  this.useAnchor_ = true;
+};
```
What are `disableMouseAnchor` and `enableMouseAnchor` for? They are not exportable (no `@api` annotation), nor are they use internally. Can we do without them?

samuellapointe commented:
I added the `@api` annotation to `enableMouseAnchor` and `disableMouseAnchor`, I had simply forgotten them. I also added a line explaining what happens when `useAnchor` is set to false in olx.js

marcjansen commented:
I'd go with only one method: `setUseAnchor(boolean)`.

If you can also add some tests, and squash the commits into one single commit, this is good to go.

Thanks!

I am currently not able to check if we have a CLA from you.

adube commented:
Samuel works at Mapgears, who has a CLA dated of 2009/03/23.   Samuel was not present at that moment.  Is this valid ?  Do we need to update our CLA ?  Should he act as individual ?  Thanks.

marcjansen commented:
IANAL, but: he says he did this contribution for work, Mapgears, so this should be fine. A personal CLA never hurts, though.

samuellapointe commented:
I put all the commits into a single one and added three small tests to check the default value of useAnchor and to test the setUseAnchor function. 

For the CLA, as mentioned before by @adube, I wasn't on the original Mapgears CLA, but I'm fine joining it or going with a personal one, whichever works the best for you.

marcjansen commented:
A very nice first contribution, @samuellapointe. Thanks.
